service/common/src/main/java/org/apache/polaris/service/http/IcebergHttpUtil.java

```diff
@@ -32,6 +32,10 @@ private IcebergHttpUtil() {}
    * @return the generated ETag
    */
   public static String generateETagForMetadataFileLocation(String metadataFileLocation) {
+    if (metadataFileLocation == null) {
+      metadataFileLocation = "";
+    }
```

Comment:
I'm not sure if we should support this case; if the metadata location is null, we can't generate a semantically meaningful etag. It might be best to error out here or refuse to build/use an etag in this case.

Reply:
Added null-checking to all callsites instead and made this throw an IllegalArgumentException. PTAL